frontend/src/app/features/details/summary/SummaryInfo.tsx

```diff
@@ -33,14 +50,57 @@ const SummaryInfo: React.FC<ISummaryInfo> = ({
       console.log('Approve Handler Not Provided');
     });
 
+  const CENTER_OF_BC: LatLngTuple = [53.7267, -127.6476];
+
+  const { data } = useMapSearch_FindSiteBySiteIdQuery({
+    variables: { siteId: siteData.id },
+  });
+
+  const selectedSiteData = data?.findSiteBySiteId.data;
+  const mapRef = useRef<Map>(null);
+  const map_fly_options = { animate: false, duration: 0.2 };
+
+  const MapWithMarker = ({ site }: { site: any }) => {
+    const map = useMap();
+
+    useEffect(() => {
+      if (!site.latdeg || !site.longdeg) return;
+      map.flyTo(
+        {
+          lat: site.latdeg,
+          lng: site.longdeg,
+        },
+        getZoom(map),
+        map_fly_options,
+      );
+    }, [site, map]);
+
+    return (
+      site && <SiteMarker position={{ lat: site.latdeg, lng: site.longdeg }} />
+    );
+  };
+
   return (
     <PanelWithUpDown
       label="Location Details"
       secondChild={
         <div className="row w-100">
           <div className="col-12 col-lg-6">
-            {/* <Map callback={() => {}} initLocation={location} readOnly={true} /> */}
+            <MapContainer
+              center={CENTER_OF_BC}
+              zoom={6}
+              zoomControl={false}
+              ref={mapRef}
+              className="map-container"
+            >
+              <TileLayer
+                attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+                url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+              />
+              {<MapWithMarker site={selectedSiteData} />}
```

Comment:
Similar situation here: what if we just place the marker like this

```tsx
            <MapContainer
              center={{ lat: siteData.latdeg, lng: siteData.longdeg }}
              // other props
            >
              <TileLayer
              // props
              />
              <SiteMarker
                position={{ lat: siteData.latdeg, lng: siteData.longdeg }}
              />
            </MapContainer>
```

This way you won't need to define the `MapWithMarker` component and won't need the map reference at all.